coalib/coala_modes.py

```diff
@@ -46,34 +46,17 @@ def mode_json(args, debug=False):
     import json
 
     from coalib.coala_main import run_coala
-    from coalib.misc.DictUtilities import inverse_dicts
-    from coalib.misc.Exceptions import get_exitcode
     from coalib.output.Logging import configure_json_logging
     from coalib.output.JSONEncoder import create_json_encoder
 
     if args.log_json:
         log_stream = configure_json_logging()
 
     JSONEncoder = create_json_encoder(use_relpath=args.relpath)
-    results = []
-
-    if args.show_bears:
-        try:
-            from coalib.parsing.FilterHelper import apply_filter
-
-            local_bears, global_bears = apply_filter(
-                'language', args.filter_by_language)
-            bears = inverse_dicts(local_bears, global_bears)
-            for bear, _ in sorted(bears.items(),
-                                  key=lambda bear_tuple:
-                                  bear_tuple[0].name):
-                results.append(bear)
-        except BaseException as exception:  # pylint: disable=broad-except
-            return get_exitcode(exception)
-    else:
-        results, exitcode, _ = run_coala(args=args, debug=debug)
 
-    retval = {'bears': results} if args.show_bears else {'results': results}
+    results, exitcode, _ = run_coala(args=args, debug=debug)
```

Comment:
Isn't now coala executed in all cases?

Reply:
Not really. In `coala.py` file, I've shifted the condition `if args.json` below `if args.show_bears` which was above that earlier. And `if.show_bears` condition always returns either 0 or some exit code. So, if `--show-bears` and `--json` are specified together, `mode_json` function won't be called. So, essentially this line won't be executed if show_bears option is specified.